modules/src/core/ics23_commitment/specs.rs

```diff
@@ -44,3 +49,26 @@ impl From<ProofSpecs> for Vec<ProtoProofSpec> {
         raw_specs
     }
 }
+
+impl From<Vec<ProofSpec>> for ProofSpecs {
+    fn from(proto_specs: Vec<ProofSpec>) -> Self {
+        let mut specs = Vec::new();
+        for ds in proto_specs {
+            // Both `ProofSpec` types implement trait `prost::Message`. Convert by encoding, then
```

Comment:
Regarding the safety of these unwraps - it is quite possible that the `ics23` crate could remove fields that are marked redundant in future versions (e.g. https://github.com/confio/ics23/blob/master/proofs.proto#L77). 
Should we use `TryFrom` instead?

Reply:
I think it would be cleaner to implement the conversion by hand rather than rely on the proto encoding, this way we would catch breaking changes at compile time rather than at runtime with `TryFrom`. What do you think?

Reply:
Apart from the gratuitous dynamism, the encode-decode roundtrip can be also a performance wart.

Reply:
Done in https://github.com/informalsystems/ibc-rs/pull/1574/commits/12a46a47e233ee6b57d8d69a8017e70d901ec759. 👍 We still need to handle the case where a new field was added to one of the proto-types and the other wasn't updated. 
Should we add an assert that makes sure the size of the structs matches (not sure that will work if the proto structs have different type layouts i.e. `repr`s)?

Reply:
I am not sure what you mean by "the case where a new field was added"? Wouldn't we catch this at compile-time whenever we update either the protos or the ics23 crate?